lib/cocoapods/sources_manager.rb

```diff
@@ -70,15 +71,27 @@ def create_source_with_url(url)
       #         The URL of the source.
       #
       def cdn_url?(url)
-        if url =~ %r{^https?:\/\/}
-          require 'typhoeus'
-
-          response = Typhoeus.get(url + '/CocoaPods-version.yml', :netrc_file => Netrc.default_path, :netrc => :optional)
-          response.code == 200 && begin
-            response_hash = YAML.load(response.body) # rubocop:disable Security/YAMLLoad
-            response_hash.is_a?(Hash) && !Source::Metadata.new(response_hash).latest_cocoapods_version.nil?
-          end
-        end
+        return unless url =~ %r{^https?:\/\/}
+
+        uri_options = {
+          :redirect => false,
```

Comment:
out of curiosity, why disable redirects?

Reply:
the previous implemention

```ruby
response.code == 200 && begin
            response_hash = YAML.load(response.body) # rubocop:disable Security/YAMLLoad
            response_hash.is_a?(Hash) && !Source::Metadata.new(response_hash).latest_cocoapods_version.nil?
          end
```

and there's a test https://github.com/CocoaPods/CocoaPods/blob/master/spec/unit/sources_manager_spec.rb#L181-L185

I just keeping the same behavior

Reply:
the linked test:
```
        it 'redirect' do
          WebMock.stub_request(:get, 'http://some_host.com/something/CocoaPods-version.yml').
            to_return(:status => 301, :body => '', :headers => { 'Location' => ['http://some_host.com'] })
          @sources_manager.cdn_url?('http://some_host.com/something').should == false
        end
```

Isn't testing that redirects are _not_ followed. It's testing that if `some_host.com/something/CocoaPods-version.yml` redirects to `some_host.com/`, then the source in question isn't a CDN. I might be misunderstanding this, but I don't think we want to disable redirects

Reply:
> It's testing that if some_host.com/something/CocoaPods-version.yml redirects to some_host.com/, then the source in question isn't a CDN.

In my understanding, `cdn_url?` attempting to get `"#{given_url}/CocoaPods-version.yml"` and trying to parse it, then it returns true.

base on your explain, you mean `cdn_url?` getting `http://some_host.com/something/CocoaPods-version.yml` through redirection returns `http://some_host.com/something`, in the test, it's not mocked, so maybe returns `SocketError`, so it's not a `cdn_url` then returns `false`

is this what your means?

Reply:
I'm not consider the logic before... my mistake, your explain sounds reasonable to me

Reply:
@amorde @segiddins 
Sorry, that's my mistake, I misunderstanding the meaning of that test,

I've corrected it

Reply:
Apologies on the unclear wording. I think that test is actually relying on a redirect, and expects redirects to be supported and not disabled. I will defer to @igor-makarov to confirm the intent behind the test

Reply:
I enable redirects now, looks CI green.

Reply:
Yes, this is the test case. Perhaps it could be clearer worded. 🤔